File: service/Stpm.Core/Entities/RankAward.cs
using Stpm.Core.Contracts;

namespace Stpm.Core.Entities;

public partial class RankAward : IEntity<int>
{
    public int Id { get; set; }

    public string AwardName { get; set; }

    public string UrlSlug { get; set; }

    public string ShortDescription { get; set; }

    public string Description { get; set; }

    public int TopicRankId { get; set; }

    public virtual ICollection<SpecificAward> SpecificAwards { get; set; } = new List<SpecificAward>();

    public virtual TopicRank TopicRank { get; set; }
}

File: client/src/app/modules/rankAward/domain/models/RankAward.ts
import { TopicRank } from "~/app/modules/topic/domain/models/TopicRank";
import { SpecificAward } from "./SpecificAward";

export interface RankAward {
  id: number;
  awardName: string;
  urlSlug: string;
  shortDescription: string;
  description: string;
  topicRankId: number;
  specificAwards: SpecificAward[];
  topicRank: TopicRank;
}
